lib/pure/memfiles.nim

```diff
@@ -39,8 +46,19 @@ type
 
 {.deprecated: [TMemFile: MemFile].}
 
+proc checkConstructed(m: var MemFile) =
+  if (not m.initialized):
```

Comment:
I still don't get the problem. Instead of testing against INVALID_HANDLE, test against initialized in the close. What else is there to do?